fluent/client/ws/connection.go

```diff
@@ -179,6 +193,10 @@ func (wsc *connection) CloseWithMsg(closeCode int, msg string) error {
 
 	wsc.closeLock.Unlock()
 
+	if wsc.log {
```

Comment:
Can we wrap it into a higher-order function 

```
type logger struct {
	Logger Logger
}

func (l *logger) Println(v ...interface{})) {
   if l.Logger == nil {
      return 
   }

  l.Logger.Println(v)
}
```

constructor 
```

wsc := &connection{
		Conn:      conn,
		closedSig: make(chan struct{}),
		connState: ConnStateOpen,
                 wsc.logger = logger{opts.Logger}
	}
```
This way you can get rid of the boolean flag and if condition checks every time.

Reply:
I thought about that. The reason I decided against it is that you're always passing the function args, even if you're not logging. That might be a micro-optimization, but it was the reason I went that way. I'll benchmark both ways.

Reply:
I went ahead and changed it. We can always optimize later